### PR TITLE
Waypoint Definition Update

### PR DESCRIPTION
**Pull request by marip8**
This PR updates the waypoint definitions:
  -  Cartesian: adds string for link to which the waypoint transform is relative
  - JointToleranced: inherits from the Joint waypoint

Corresponds to `trajopt_ros` PR [130](https://github.com/ros-industrial-consortium/trajopt_ros/pull/130)

### CODE REVIEW

File: tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/core/waypoint.h [hunk 1]
```diff
@@ -189,48 +194,44 @@ class CartesianWaypoint : public Waypoint
...
     Eigen::Quaterniond q(cartesian_position_.rotation());
     return Eigen::Vector4d(q.w(), q.x(), q.y(), q.z());
   }
 
+  /** @brief Gets the name of the link to which this position is relative */
+  std::string getParentLinkName() const { return parent_link_; }
+
 protected:
   Eigen::Isometry3d cartesian_position_; /**< @brief Pose of this waypoint */
+  std::string parent_link_; /** @brief The link to which this position is defined */
```
Comment (Levi-Armstrong) [inline tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/core/waypoint.h]: I would change doxygen to `/**< @brief`

**Comment by Levi-Armstrong** [comment]
Also CI has two failures in tesseract_ros_examples.

**Comment by marip8** [comment]
@Levi-Armstrong as mentioned in the description, this PR depends on `trajopt_ros` [PR 130](https://github.com/ros-industrial-consortium/trajopt_ros/pull/130). I'll restart the build for this PR once the TrajOpt one is merged, otherwise the tests will continue to fail

**Comment by Levi-Armstrong** [comment]
@marip8 I created a PR against your branch updating the trajopt planners to set the target based on the cart waypoint parent link.

Also, Descartes needs to be updated to use the cart waypoint parent link name.

**Comment by marip8** [comment]
@Levi-Armstrong the changes you provided in that PR are addressed in #126 where I added the same changes to the utility that creates the `trajopt::TermInfo` for Cartesian waypoints. My suggestion regarding your PR is to hold off because both of those planners your PR modifies were removed in #126.

**Pull request merged**